ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/PrivGetFilterChanges.java

```diff
@@ -53,7 +57,8 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
     final String privacyGroupId = requestContext.getRequiredParameter(0, String.class);
     final String filterId = requestContext.getRequiredParameter(1, String.class);
 
-    checkIfPrivacyGroupMatchesAuthenticatedEnclaveKey(requestContext, privacyGroupId);
+    final BlockParameter blockParameter = filterManager.getToBlock(filterId);
```

Comment:
If `filterManager.getToBlock` can return null, we need to be careful when using the `blockParameter` variable in `getBlockNumberToCheckForGroupMembership `

Reply:
fixed